sql/catalyst/src/main/scala-2.12/org/apache/spark/sql/catalyst/util/CaseInsensitiveMap.scala

```diff
@@ -52,6 +52,8 @@ class CaseInsensitiveMap[T] private (val originalMap: Map[String, T]) extends Ma
   override def -(key: String): Map[String, T] = {
     new CaseInsensitiveMap(originalMap.filter(!_._1.equalsIgnoreCase(key)))
   }
+
+  def toMap: Map[String, T] = originalMap
```

Comment:
shall we use consistent name `asCaseSensitiveMap`?

Reply:
In that case, among all the existing `extraOptions.toMap`s, do I need to rename only `AppendData.byName` instance?
```
sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:    val params = extraOptions.toMap ++ connectionProperties.asScala.toMap
sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:      extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:      extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:              AppendData.byName(relation, df.logicalPlan, extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:                relation, df.logicalPlan, Literal(true), extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:                  extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:        options = extraOptions.toMap).planForWriting(mode, df.logicalPlan)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:        AppendData.byPosition(table, df.logicalPlan, extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:          OverwritePartitionsDynamic.byPosition(table, df.logicalPlan, extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:          OverwriteByExpression.byPosition(table, df.logicalPlan, Literal(true), extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:        AppendData.byName(v2Relation, df.logicalPlan, extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:          extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:          extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala:    val storage = DataSource.buildStorageFormatFromOptions(extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala:      options = extraOptions.toMap)
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala:        extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala:        extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala:        extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala:        extraOptions.toMap,
sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala:      options = extraOptions.toMap,
```

Reply:
that's too many. let's keep `toMap` then